db/dbtypes/types.go

```diff
@@ -549,3 +567,87 @@ type BlockStatus struct {
 	Hash        string `json:"hash"`
 	NextHash    string `json:"next_hash"`
 }
+
+// syncStatus makes it possible to update the user on the progress of the
+// blockchain db syncing that is running in the background after new blocks
+// were detected on system startup.
+type syncStatus struct {
+	sync.RWMutex
+	Update map[string]*SyncStatusInfo
+}
+
+// SyncStatusInfo defines information for a single update type.
+type SyncStatusInfo struct {
+	// percentage change
+	Change float64
+	// used to display the main message about the currect sync.
+	Msg string
+	// displays any other information about the current main sync. This value
+	// may include but not limited to; db indexing, deleting duplicates etc.
+	OtherMsg string
+	// defines estimated time to completing the sync
+	Time       int64
+	UpdateType string
+}
+
+// SyncStatusUpdate defines a thread-safe way to update the sync status updates.
+func SyncStatusUpdate(from, to, timeToComplete int64, updateType, msg string) {
+	blockchainSyncStatus.Lock()
+	defer blockchainSyncStatus.Unlock()
+
+	percentage := 0.0
+	if to > 0 {
+		percentage = math.Floor((float64(from)/float64(to))*10000) / 100
+	}
+
+	val := &SyncStatusInfo{
+		Change:     percentage,
```

Comment:
I'm trying to understand the semantics of the input arguments to `SyncStatusUpdate`.

When this function is called [here](https://github.com/decred/dcrdata/pull/710/files#diff-60206eba7205ea27cd7a8137d424f61cR2018), `from` is set to `int64(i)`, while `to` is set to `int64(totalVinIbIDs)`, where `i` is the current block and `totalVinIbIDs` is the number of blocks to process.  So this is a measure of total progress, not an incremental change.

The definition of `percentage` above confirms that it's "percentage complete" (as opposed to "percentage remaining" or "percentage change since last update" etc.).

So unless I'm misunderstanding, please rename `Change` to `PercentComplete` for clarity.